rest/changes_api_test.go

```diff
@@ -601,6 +602,157 @@ func TestChangesLoopingWhenLowSequence(t *testing.T) {
 
 }
 
+func TestConcurrentDelete(t *testing.T) {
```

Comment:
Run this test in a loop to make it more likely to hit the race condition in a given run.